### PR TITLE
Improve documentation for framework.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/screen.dart [hunk 1]
```diff
@@ -145,6 +145,17 @@ enum ScreenMetaData {
...
 
 /// Defines a page shown in the DevTools [TabBar].
+///
+/// A devtools screen can be in three modes: offline-data, connected
+/// and not-connected. The screen can support only subset of modes.
+///
+/// For offline-data and connected modes:
+/// * Override [Screen.buildScreenBody] to build content
+/// * Use [ProvidedControllerMixin] to access controller
+/// * See [OfflineScreenControllerMixin] on how enable and handle offline-data mode.
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/screen.dart:155]: "See [OfflineScreenControllerMixin] for documentation on how to enable and handle offline-data mode for a screen."

File: packages/devtools_app/lib/src/shared/screen.dart [hunk 2]
```diff
@@ -145,6 +145,17 @@ enum ScreenMetaData {
 }
 
 /// Defines a page shown in the DevTools [TabBar].
+///
+/// A devtools screen can be in three modes: offline-data, connected
+/// and not-connected. The screen can support only subset of modes.
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/screen.dart:150]: what do you mean by "The screen can support only subset of modes"?

Comment (polina-c) [reply packages/devtools_app/lib/src/shared/screen.dart:150]: Not all screens support all modes :)

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/screen.dart:150]: got it. Maybe rephrase for clarity? "A screen may support any combination of modes."

**Review by kenzieschmoll — APPROVED** [review]
